### ISSUE TITLE
Documentation of subType issues (esp in relation to attendance)

### ISSUE THREAD

**Issue by arc12** [jiscdev/xapi#196]
I think that a series of discussions and decisions relating to VLE have left us with some unclear and/or contradictory documentation.

On this page - https://github.com/jiscdev/xapi/blob/1.0/vocabulary.md#activity-types - it says:
"Some examples of subTypes are included here. However, the list is not exhaustive, and it is the responsibility of the plugin to define the subType."

I think this should be tightened to make it clear that this statement is referring to the "Example Subtypes" components of the various tables. i.e. that none of the subheadings are sub-types, all are activity types. I think we should also indicate that sub-types are not defined in the recipes at all, i.e. its not so much that the lists are not exhaustive, they are purely indicative.

Now moving on to the attendance page: https://github.com/jiscdev/xapi/blob/1.0/recipes/attendance/attendance.md

A. the description for subType in the table for Object properties states:
"May be used to qualify what kind of timetabled event occurred using an IRI given on the vocabulary page.".

I think this is incorrect, as subType is for application-specific nuance.  I think this is a vestige of earlier work (see below, under B, where I use "vestige")

B. this lacks specific guidance on suitable values of object.definition.type; the reader is directed to the vocab page where they find: http://activitystrea.ms/schema/1.0/event, http://xapi.jisc.ac.uk/lecture, and http://id.tincanapi.com/activitytype/tutor-session. 

I think the documentation should clearly state what values are permitted for attendance statements. I think some narrative would also be useful (outside the table for Object) which explains how object.definition.type and the http://xapi.jisc.ac.uk/event_timetabled are intended to be used together in typical scenarios. (and possibly the Purpose section should be explicit that the statement template does NOT cover pastoral sessions OR it should be explicit that it DOES cover them - i.e. we need to clarify whether a "learning activity" is scoped to subject matter or not, since it isn't accurate to assume that learning is only subject specific)

This brings me to what I think is a vestige of earlier work. I suspect that the lecture and tutor-session activity types were originally intended to be subTypes when we had an object.definition.type of (something like) "timetabled event". Now subtype is purely off-spec, I think we've not got the right list of activity types, and suggest they should be: http://activitystrea.ms/schema/1.0/event (to cover generic events), http://xapi.jisc.ac.uk/class (a group teaching event), and http://id.tincanapi.com/activitytype/tutor-session (a solo, or very small group, teaching event). The particular problem with the current set is that "Lecture" is not sufficiently generic for object definition type for a fair proportion of timetabled events, which could lead to (e.g.): some people just use lecture anyway, some people use http://activitystrea.ms/schema/1.0/event and subtypes for all events, some people use http://activitystrea.ms/schema/1.0/event for non-lectures and http://xapi.jisc.ac.uk/lecture for lectures. ARGHHH

Bottom line: the spec needs to be clear enough that when someone looks at different sets (from different sources) of data generated against it, they stand a fair chance of getting the same usage of object.definition.type, http://xapi.jisc.ac.uk/event_timetabled, and http://xapi.jisc.ac.uk/subType.

C. event_type_id appears to be redundant if we have possibilities object.definition.type which differentiate event types + the possibility to use subType. Two places to put basically the same info, with both being optional, is a recipe for needless variation in data, which places a burden on data consumers yet has no value to data suppliers.

I suggest event_type_id is removed in favour of subType.

**Comment by ds10** [comment]
I've tweaked the vocab doc because it wasn't very clear that we were talking about subTypes and not Activity types in that sentence you've highlighted.

However, I'll hold off the attendence statement template changes and add this to the weekly call to decide what we want to do.

**Comment by arc12** [comment]
I think this is largely a 1.0.1 issue as it relates to attendance.
Could we modify the milestone accordingly and create a new issue as required for 1.1 aspects

**Comment by ds10** [comment]
I've set the milestone to 1.0.1 - I don't think we unpicked this properly at the last call. I'll hold off any codebase changes until then

**Comment by ds10** [comment]
Weekly call: Adam to fork and edit some solutions

**Comment by arc12** [comment]
PR just made.
Approach has been to simplify to the core, but I hope without loosing _necessary_ expressiveness and clarity in query/aggreagation.

subType is now "the" place to distinguish between kinds of timetabled event, with no specified vocab within

**Comment by alanepaull** [comment]
Looking at logging in and logging out: we currently have object.definition.type of http://activitystrea.ms/schema/1.0/application. Then we're supposed to use subType to indicate the application logged into. In vocabulary, we only have http://id.tincanapi.com/activitytype/mobile-app listed, but there's also http://id.tincanapi.com/activitytype/lms mentioned in an example. I'm sure there must be others. Is it worth collecting them and listing in the vocab?

So for now, that would give subTypes of http://activitystrea.ms/schema/1.0/application for logging in / out:

http://id.tincanapi.com/activitytype/mobile-app
http://id.tincanapi.com/activitytype/lms

We could add others as they arise?

Alan

**Comment by alanepaull** [comment]
At present we don't have a specific object.definition.type in the vle_resource_viewed statement template (https://github.com/jiscdev/xapi/blob/1.0.1/recipes/vle/Module-View.md). This means there is, or is likely to be, confusion between values for object.definition.type and subType.

Similar to the approach used by @arc12 re attendance object.definition.type for timetabled events, we could state in vle_resource_viewed that object.definition.type must be "http://xapi.jisc.ac.uk/vle/content" and define the resource in more detail in the subType. We have some subType examples for this in the vocabulary file (see "VLE content").

Alan

**Comment by alanepaull** [comment]
Commit https://github.com/jiscdev/xapi/commit/326b08038c708d879fcbba70fb47cca0ab92b966 shows update as previous message.

**Comment by alanepaull** [comment]
Monday 8 Jan 18: review object definition types in all statement templates, and their associated subTypes.

**Comment by ds10** [comment]
updated the  documentation to make it clear when a statment template uses a type
